src/game.h

```diff
@@ -1042,6 +1048,7 @@ class game
         time_point nextweather; // The time on which weather will shift next.
         int next_npc_id, next_mission_id; // Keep track of UIDs
         std::map<mtype_id, int> kills;         // Player's kill count
+        std::map<species_id, int> species_kills;
```

Comment:
This seems completely redundant and therefore unnecessary. You can get the kill count by summing up all the kills of those monsters that belong to the species. In other words: iterate over `kills` and, if the monster belongs to the species in question, add the count to a total and return that.

Reply:
> iterate over kills and, if the monster belongs to the species in question, add the count to a total and return that.

But `kills` returns kills for the monster in question, we'll still need another variable for storing species kill, or I'm wrong?

Reply:
> But kills returns kills for the monster in question, we'll still need another variable for storing species kill, or I'm wrong?

If the player has killed 4 "mon_zombie", 7 "mon_dog" and 2 "mon_boomer", they have kill 6 monster of species "ZOMBIE" (4 zombie + 2 boomer). Iterate over the `kills` map, inspect the monster type to see if it belongs to the species you are interested in (`mtype::in_species`) and (if so) add up the kill count and return the final sum.

Reply:
```C++
int game::kill_count( const species_id &spec )
{
    int result = 0;
    for( const auto &pair : kills ) {
        if( pair.first->in_species( spec ) ) {
            result += pair.second;
        }
    }
    return result;
}
```

Reply:
I got a feeling that either I'm not describing the problem clearly enough or I'm too dumb to understand your idea.
You rewrote the whole `kill_count` function, so now it counts total number of all killed monsters of a given species, right? But if so, we thus lost the ability to count number of killed monsters of a given type, right?

Reply:
This is C++, you can have multiple functions with the same name as long as the types of the parameter differ.
`int game::kill_count( mtype_id )` and `int game::kill_count( species_id )` are two completely different functions.